Firestore/core/src/firebase/firestore/remote/remote_objc_bridge.mm

```diff
@@ -127,170 +112,233 @@ bool IsLoggingEnabled() {
 
 // WatchStreamSerializer
 
-GCFSListenRequest* WatchStreamSerializer::CreateWatchRequest(
-    const QueryData& query) const {
-  GCFSListenRequest* request = [GCFSListenRequest message];
-  request.database = [serializer_ encodedDatabaseID];
-  request.addTarget = [serializer_ encodedTarget:query];
-  request.labels = [serializer_ encodedListenRequestLabelsForQueryData:query];
-  return request;
+WatchStreamSerializer::WatchStreamSerializer(Serializer serializer)
+    : serializer_{std::move(serializer)} {
 }
 
-GCFSListenRequest* WatchStreamSerializer::CreateUnwatchRequest(
-    TargetId target_id) const {
-  GCFSListenRequest* request = [GCFSListenRequest message];
-  request.database = [serializer_ encodedDatabaseID];
-  request.removeTarget = target_id;
-  return request;
+Message<google_firestore_v1_ListenRequest>
+WatchStreamSerializer::CreateWatchRequest(const QueryData& query) const {
+  Message<google_firestore_v1_ListenRequest> result{
+      google_firestore_v1_ListenRequest_fields};
+  auto& request = result.mutable_proto();
+
+  request.database = serializer_.EncodeDatabaseId();
+  request.which_target_change =
+      google_firestore_v1_ListenRequest_add_target_tag;
+  request.add_target = serializer_.EncodeTarget(query);
+
+  auto labels = serializer_.EncodeListenRequestLabels(query);
```

Comment:
Ideally, `Serializer` should return a Nanopb type, but in this case, I'd postpone it until I implement `ArrayRef` (so that serializer can return both the array and its size in a single value).